oneof.go

```diff
@@ -52,6 +56,12 @@ func (o *oneof) Descriptor() *descriptor.OneofDescriptorProto { return o.desc }
 func (o *oneof) Message() Message                             { return o.msg }
 func (o *oneof) setMessage(m Message)                         { o.msg = m }
 
+func (o *oneof) IsSynthetic() bool {
+	return o.Syntax() == Proto3 &&
+		len(o.flds) == 1 &&
+		!o.flds[0].InRealOneOf()
```

Comment:
nit: may i recommend inlining `!f.desc.GetProto3Optional()` here. Just makes it more immediately clear what's going on.

Reply:
Hmm, I think using `.GetProto3Optional()` (which is an explicit property of  a `Field`) outside the context of a field will be more confusing over time.

From the perspective of a Synthentic OneOf, I think it reads fine right?
- Syntax is Proto3
- OneOf contains exactly one field (an extra semantic check)
- The singular field is not considered to be in a `RealOneOf`

WDYT?